packages/commerce-sdk-react/src/components/ShopperExperience/Component/index.tsx

```diff
@@ -20,17 +22,23 @@ const ComponentNotFound = ({typeId}: ComponentType) => (
  *
  * @param {PageProps} props
  * @param {Component} props.component - The page designer component data representation.
+ * @param {string} [props.code] - The raw JSW code for the component.
  * @returns {React.ReactElement} - Experience component.
  */
-export const Component = ({component}: ComponentProps) => {
+export const Component = ({component, code}: ComponentProps) => {
     const pageContext = usePageContext()
-    const ComponentClass = pageContext?.components[component.typeId] || ComponentNotFound
     const {data, ...rest} = component
+    let instance = <ComponentNotFound {...rest} {...data} />
+    if (code) {
+        instance = <JsxParser components={pageContext?.jsxParserComponents} jsx={code} />
```

Comment:
@kevinxh feels weird that this doesn't accept passthrough props a la `{...rest}` like the other component does. 🤔

Reply:
@kevinxh I think we want to allow-list components and have that allow-list in our control. Version 1 of this feature does not include customers being able to customize which Chakra UI components the LLM uses in it's code generation. Allowing the user to set it here gives the wrong impression that it's within the user's control

Reply:
Passed `{...rest}` and implemented warnings. Due to separation of concerns, it is best to keep Chakra a dependency of `retail-react-app` and not leak into `commerce-sdk-react`. 

A allow list is temporarily implemented as an array of strings in the code. we can re-visit this later.